forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -178,57 +186,105 @@ pub async fn deploy(command: cmd::Deploy) -> Result<Vec<DeployedContract>> {
+            (None, false) => rand::random(),
+            (Some(_), true) => {
+                bail!("Both `--salt` and `--default-salt` were specified: must choose one")
+            }
+        };
+        let contract_id = deploy_pkg(
+            &command,
+            pkg,
+            salt,
+            &provider,
+            &signing_key,
+            node_url.clone(),
+        )
+        .await?;
+        contract_ids.push(contract_id);
+    }
+    Ok(contract_ids)
+}
+
+/// Prompt the user to confirm the transactions required for deployment, as well as the signing key.
+async fn confirm_transaction_details(
+    pkgs_to_deploy: &[&Arc<BuiltPackage>],
+    command: &cmd::Deploy,
+    node_url: String,
+) -> Result<(Provider, SecretKey)> {
+    // Confirmation step. Summarize the transaction(s) for the deployment.
+    let tx_summary = pkgs_to_deploy
+        .iter()
+        .map(|pkg| format!("deploy {}", pkg.descriptor.manifest_file.project_name()))
+        .collect::<Vec<_>>()
+        .join(" + ");
+
+    let tx_count = pkgs_to_deploy.len();
+
+    println_action_green("Confirming", &format!("transactions [{tx_summary}]"));
+    println_action_green("", &format!("Network: {node_url}"));
+
+    let provider = Provider::connect(node_url.clone()).await?;
+
     let wallet_mode = if command.default_signer || command.signing_key.is_some() {
         WalletSelectionMode::Manual
     } else {
-        let password = prompt_forc_wallet_password(&default_wallet_path())?;
+        println_action_green("", &format!("Wallet: {}", default_wallet_path().display()));
+        let password = prompt_forc_wallet_password()?;
         WalletSelectionMode::ForcWallet(password)
     };
 
-    for pkg in built_pkgs {
-        if pkg
-            .descriptor
-            .manifest_file
-            .check_program_type(&[TreeType::Contract])
-            .is_ok()
-        {
-            let salt = match (&contract_salt_map, command.default_salt) {
-                (Some(map), false) => {
-                    if let Some(salt) = map.get(pkg.descriptor.manifest_file.project_name()) {
-                        *salt
-                    } else {
-                        Default::default()
-                    }
-                }
-                (None, true) => Default::default(),
-                (None, false) => rand::random(),
-                (Some(_), true) => {
-                    bail!("Both `--salt` and `--default-salt` were specified: must choose one")
-                }
-            };
-            let contract_id = deploy_pkg(
-                &command,
-                &pkg.descriptor.manifest_file,
-                &pkg,
-                salt,
-                &wallet_mode,
-            )
-            .await?;
-            contract_ids.push(contract_id);
-        }
-    }
-    Ok(contract_ids)
+    // TODO: Display the estimated gas cost of the transaction(s).
+    // https://github.com/FuelLabs/sway/issues/6277
```

Comment:
Nice was already going to say this would be really handy to have in this workflow